src/utils/getNodeType.ts

```diff
@@ -0,0 +1,6 @@
+import { JSONSchema4, JSONSchema4TypeName } from 'json-schema';
+import { inferType } from './inferType';
+
+export function getNodeType(node: JSONSchema4): JSONSchema4TypeName | JSONSchema4TypeName[] | undefined {
```

Comment:
Let's just extend `inferType` function.

Reply:
That was actually my first intention but thought it might not be precise enough. 
I'll move it there. But if so shouldn't `type` be checked before `properties` and `items`?

Reply:
Yeah, it could be moved to the top.